internal/processing/processors/v2alpha1/authorizationpolicy/processor.go

```diff
@@ -12,11 +12,13 @@ import (
 )
 
 // NewProcessor returns a Processor with the desired state handling for AuthorizationPolicy.
-func NewProcessor(log *logr.Logger, rule *gatewayv2alpha1.APIRule) Processor {
+func NewProcessor(log *logr.Logger, rule *gatewayv2alpha1.APIRule, oryPassthrough bool) Processor {
```

Comment:
I don't like that we now have to add migration code to the "normal" v2alpha1 code, but I understand that there is no good way to do it different. Maybe it already helps with separation if we have a different helper function to create a migration processor:

```go
// NewProcessor returns a Processor with the desired state handling for AuthorizationPolicy.
func NewProcessor(log *logr.Logger, rule *gatewayv2alpha1.APIRule) Processor {
	return Processor{
		apiRule: rule,
		creator: creator{},
		Log:     log,
	}
}

// NewMigrationProcessor returns a Processor with the desired state handling for AuthorizationPolicy during the migration process from v1beta1 to v2alpha1.
func NewMigrationProcessor(log *logr.Logger, rule *gatewayv2alpha1.APIRule, oryPassthrough bool) Processor {
	return Processor{
		apiRule: rule,
		creator: creator{
			oryPassthrough: oryPassthrough,
		},
		Log: log,
	}
}
```